### ISSUE TITLE
[BUG] Potential bug when do batch inference

### ISSUE THREAD

**Issue by CiaoHe** [haotian-liu/LLaVA#575]
Hi, 
I tried to modify the code for batch inference but found one bug,

In the `prepare_inputs_labels_for_multimodal` if we provide attention_mask but not given labels, then when execute L187, the  local variable `_new_labels` referenced before assignment, 
https://github.com/haotian-liu/LLaVA/blob/539d666848245a7075fb485cee5706e68e0489d2/llava/model/llava_arch.py#L187-L191

So I guess maybe we need to change the logic as L199-L202 did, that pad attention_mask just dependent on the `new_input_embeds.shape[1] - input_ids.shape[1]`

Best,

**Comment by haotian-liu** [comment]
Hi, we have rewritten the multimodal processing function and this should be addressed. Also, I am trying to work on an efficient batch inference. Closing this and consolidating the discussion to https://github.com/haotian-liu/LLaVA/issues/754. Thanks.
